### PR TITLE
Antlers Runtime Parity and Bug Fixes

### LINKED ISSUE #5545: Variables should override tags (Antlers Runtime)
### Bug description

The link fieldtype with the runtime enabled, seems to spit out null instead of the link. If you do a dump, you can see though, that the value is present (inside a value object).

### How to reproduce

Create a link fieldtype (my handle is ``link``), and see that it works as expected with the regex parser. As soon as you switch over to the runtime, it doesn't work.

### Logs

_No response_

### Versions

Statamic 3.3.1 Pro
Laravel 9.5.1
PHP 8.0.13
No addons installed

### Installation

Existing Laravel app

### Additional details

_No response_

### LINKED ISSUE #5548: Antlers Stacks (v3.3) could use a more helpful error message when used in the wrong direction
### Bug description

When trying to use the new stacks feature, I kept running into the following error:

```
Undefined array key "<a UUID that changed with each request>"
```

I narrowed the problem down to the new stacks feature. It seems that stacks defined and/or referenced in a different file than their defined throws this error. Which, unfortunately, makes the feature practically useless at the moment. :(

### How to reproduce

1. Check out this repo: https://github.com/theutz/stack-example 
2. Look at the `home.antlers.html` and `layout.antlers.html` file.
    - The goods are at this commit: https://github.com/theutz/stack-example/commit/267c61894b57d4e97c3ccaaa751cce93f7ccea01
3. Run the site and enjoy the fancy Ignition exception screen. :)

### Logs

_No response_

### Versions

Statamic 3.3.1 Solo
Laravel 9.5.1
PHP 8.1.2
No addons installed

### Installation

Fresh statamic/statamic site via CLI

### Additional details

_No response_

### LINKED ISSUE #5554: Problem with depth in nested navigation with new Antlers parser
### Bug description

The new `runtime` Antlers parsers seems to trip up when using `{{ depth }}` in `{{ nav }}` tags.

### How to reproduce

On a fresh installation of Statamic into an existing Laravel app, create navigation that has nested levels:

```
tree:
  -
    id: 7c15e122-9057-483f-82d9-2e481431c5bf
    title: Home
    url: /
  -
    id: d5e0312d-2c9d-4746-aeed-ace069a6331a
    title: About
    url: /about
    children:
      -
        id: 52486fc9-b997-48af-b57f-ecdd56cd840e
        title: Team
        url: /about/team
        children:
          -
            id: 73f82f54-78df-4987-90e6-d5ded3020a45
            title: Leadership
            url: /about/team/leadership
  -
    id: 94f69c31-237b-4237-ba55-177ef0929449
    title: Projects
    url: /projects
    children:
      -
        id: 9b3a4919-40b0-4a3b-9b8b-44a427d16c07
        title: Supernova
        url: /projects/supernova
      -
        id: c5ab4d57-d53b-429b-ae50-7b1086cc8ab0
        title: Meadow
        url: /projects/meadow
        children:
          -
            id: e602b108-701d-403b-a206-ce86449995c7
            title: Discussion
            url: /projects/meadow/discussion
  -
    id: 9ca8bf85-089c-4f97-b449-2230eb21fc5d
    title: Contact
    url: /contact
```

Now set up the following in a view:

```
{{ nav:main include_home="true" }}
    <div>{{ depth }} {{ title }}</div>
    {{ if children }}
        {{ *recursive children* }}
    {{ /if }}
{{ /nav:main }}
```

Using the old `regex` Antlers parser, the output is what you would expect:

```
1 - Home
1 - About
2 - Team
3 - Leadership
1 - Projects
2 - Supernova
2 - Meadow
3 - Discussion
1 - Contact
```

But under the new `runtime` parser:

```
1 - Home
1 - About
2 - Team
3 - Leadership
1 - Projects
1 - Supernova
1 - Meadow
2 - Discussion
1 - Contact
```

### Logs

_No response_

### Versions

Statamic 3.3.1 Solo
Laravel 9.5.1
PHP 8.1.3
No addons installed

### Installation

Existing Laravel app

### Additional details

_No response_

### LINKED ISSUE #5560: Collection from modifier not found 3.3.1
### Bug description

Error after update to 3.3 with runtime parser. With regex it's working.

### How to reproduce

View:
```
{{ collection from="{collection}" limit="12" offset="5" sort="date:desc" as="links" }}
```

Error:
```
Statamic\Modifiers\ModifierNotFoundException
Modifier [from] not found
```

### Logs

_No response_

### Versions

Statamic 3.3.1 Pro
Laravel 8.83.5
PHP 8.0.12

### Installation

Fresh statamic/statamic site via CLI

### Additional details

_No response_

### LINKED ISSUE #5566: Aliasing entries of collection in runtime antlers throws an exception
### Bug description

I have an entries fieldset "contacts" of the collection Contacts:

```
  -
    handle: contacts
    field:
      mode: typeahead
      create: true
      collections:
        - contacts
      display: Contacts
      type: entries
      icon: entries
```

When entries is alised I'm getting "**_Call to a member function all() on array_**" in runtime mode. Regex mode works as expected.

```
{{ contacts as="contact_items" }}
    {{ if !no_results }}
        <div class="block">
            <div class="block__container">
                {{ contact_items }}
                    {{ partial:partials/contact_locket }}
                {{ /contact_items }}
            </div>
        </div>
    {{ /if }}
{{ /contacts }}
```

### How to reproduce

- runtime mode version set in antlers config
- use entries pair tag with an aliasing inside the antlers template (see above)

### Logs

_No response_

### Versions

Statamic 3.3.x-dev Pro
Laravel 8.83.5
PHP 8.1.1
jacksleight/bard-mutator 0.1.4
octoper/statamic-html-minify 0.0.5

### Installation

Starter Kit using via CLI

### Additional details

_No response_

### LINKED ISSUE #5570: `taxonomy` variable causes exception
### Bug description

Having a variable named `taxonomy` will throw:

```
Statamic\Exceptions\TaxonomyNotFoundException Taxonomy [] not found
```

### How to reproduce

Inject a variable named `taxonomy` and attempt to use it as a string `{{ taxonomy }}`.

### Logs

_No response_

### Versions

Statamic 3.3.1 Pro
Laravel 8.83.5
PHP 8.0.12

### Installation

Fresh statamic/statamic site via CLI

### Additional details

Extracted from https://github.com/statamic/cms/issues/5560

### LINKED ISSUE #5611: Variable Assignment Inside Nested Tags Not Working
### Bug description

Incrementing values inside nested arrays/tags does not increment values as expected.

### How to reproduce

Attempt to increment a value from nested arrays/tags:

```
{{ data_count = 0 }}

{{ array_one }}
    {{ array_two }}
        {{ data_count += 1 }}
    {{ /array_two }}
{{ /array_one }}
```

### Logs

_No response_

### Versions

Statamic 3.3.2 Solo
Laravel 9.5.1
PHP 8.1.3
No addons installed

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by JohnathonKoster**
This PR provides additional parity when compared to Regex behavior, and resolves a few bugs. The most notable change is that this PR flips the behavior so that variables are prioritized over tags.

117829200-Uncle-Buck-quotes-2

Related to the variable over tag prioritization change:

* Closes #5570 
* Closes #5560 
* Closes #5545 

Additional bug fixes include:

* Closes #5566 - No longer blindly attempts to call `all` when builders are being aliased
* Closes #5554  - Decrements the active depth when moving back up to a parent's depth instead of resetting to root each time
* Closes #5548  - Corrects array offset error. Upon further review, being able to push to stacks before they are defined seemed useful, so this behavior is now supported. Additionally, the Runtime will now adjust it's behavior when rendering layout vs. content templates to accommodate this (and other "stack like" features).
* Corrects an issue where using the `{{ $var }} {{ /$var }}` syntax on query builders would skip the `{{ query:var }}` tag interception
* Correctly restores the data manager's paired/non-paired state after resolving variables with modifier parameters (https://github.com/statamic/cms/issues/5545#issuecomment-1075610427)
* Closes #5611  - Improves how assignments are traced when moving between nested loops and tag pair contexts

Improvements:

* Resolves dynamic bindings inside parameter-style modifiers: `{{ variable :modifier="other_variable" }}`

### CODE REVIEW

**Comment by jasonvarga** [comment]
Just dropping this here to keep track of the review of all the tagged issues.

- [ ] #5570 
- [ ] #5560
- [ ] #5545 
- [ ] #5566 
- [ ] #5554 
- [ ] #5548 
- [ ] `{{ $var }} {{ /$var }}`
- [ ] `:modifier`

File: tests/Antlers/Runtime/VariablePriorityTest.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public function test_strings_can_be_used_inside_tags_with_similar_names()
+    {
+        $template = <<<'EOT'
+{{ collection from="{collection}" }}{{ /collection }}
+EOT;
+
+        $this->expectException(CollectionNotFoundException::class);
+        $this->expectExceptionMessage('Collection [news] not found');
+        $this->renderString($template, ['collection' => 'news'], true);
+    }
```
Comment (jasonvarga) [inline tests/Antlers/Runtime/VariablePriorityTest.php:67]: This feels so magical. (But expected) How does it know you aren't intending to call a `from` modifier on a variable named `collection`?

**Comment by jasonvarga** [comment]
> Improvements:
>
> Resolves dynamic bindings inside parameter-style modifiers: `{{ variable :modifier="other_variable" }}`

It looks like this already worked.
However it seems that this PR supports putting modifiers inside the parameter (e.g. `{{ variable :modifier="other_variable | upper" }}`

File: tests/Antlers/Runtime/VariablePriorityTest.php (hunk 1 above)
Comment (JohnathonKoster) [reply tests/Antlers/Runtime/VariablePriorityTest.php:67]: The NodeProcessor now utilizes the following simplified steps to determine if it should process something as a tag or not. If a template author has not forced the item to be a variable using the `$` syntax, or forced it to be a tag by using the `%` syntax, the processor will perform the following checks. In the following, the node's default behavior is determined ahead of time by checking if the node's name matches a registered tag (and is exposed via the `$node->isTagNode` property).

1. Is the variable part of a tag pair? If so, and the resolve value is loopable, we proceed as a variable. If the resolved value is not loopable, we proceed based on the node's default behavior.
2. If the variable is not a tag pair, does the variable existing within the current context? If so, we proceed as a variable. If not, we proceed with the node's default behavior.

There are some additional checks surrounding query builder instances to ensure that they are sent to the `query` tag (unless they appear as part of something like `{{ query_builder.0.title }}` - in these scenarios, the value is simply resolved and the rest of the variable path is processed like any other array).

**Comment by JohnathonKoster** [comment]
> 

Correct - should have clarified that better.

**Pull request merged**